### ISSUE TITLE
Please add a maxRange option for xAxis

### ISSUE THREAD

**Issue by mlvn23** [highcharts/highcharts#6507]
This is mainly for HighStock. We load data that spans decades, and we'd like to allow our user to be able zoom out at a max of 5-year (for example) **sliding window**, since plotting more than 5 years with OHLC and Candlestick is both counter-productive and CPU-intensive. This way, the user can still explore the entire data history, but only in 5-year or less window.

Automatic data grouping in financial charts are not recommended, since the user normally wants to analyze the chart patterns and use indicators on a particular time-frame. So a daily chart cannot suddenly plot weekly bars without user command.

From the navigator perspective, when the user drags the panel to the left handle, it should allow it to expand up to the maxRange without moving the right handle (and vice-versa). Once it reaches the maxRange, both the left and the right would have to move at the same time to maintain the maxRange, creating a "sliding window" effect (similar to what happens when you drag the navigator in the middle - both handles move together).

**Comment by KacperMadej** [comment]
Related userVoice request: https://highcharts.uservoice.com/forums/55896-highcharts-javascript-api/suggestions/4120781-maxrange-for-axes

**Comment by sebastianbochan** [comment]
_Workaround_
Catch the afterSetExtremes event and correct extremes.

Demo:
- http://jsfiddle.net/B7vCR/3/

**Comment by mlvn23** [comment]
Thanks for the work-around. It should be useful in the interim. 

I tried to make it not right-biased so that the max gets adjusted if the user moves the left handle, and min gets adjusted if the user moves the right handle. I did it by remembering the previous extremes. It somewhat works, but apparently the afterSetExtremes are not called if the user uses the range selector or if the user moves the navigator using the the scroll bar.

Hopefully, we can get a more integrated / easier solution. Something that would integrate well with the Range Selectors and the scrollbar similar to minRange.
